recurring_contract/model/recurring_contract.py

```diff
@@ -124,8 +124,7 @@ def _get_total_amount(self, cr, uid, ids, name, args, context=None):
             states={'draft': [('readonly', False)]},
             track_visibility="onchange"),
         'end_date': fields.date(
-            _('End date'), readonly=False,
-            states={'terminated': [('readonly', True)]},
+            _('End date'), readonly=True,

```

Comment:
You can make this change in module sponsorship_compassion instead of module recurring_contract.